chall02/jcervill.py

```diff
@@ -0,0 +1,82 @@
+import sys
+
+i = 0
+morse = {
+	'a':'.-',
+	'A':'.-',
+	'b':'-...',
+	'B':'-...',
+	'c':'-.-.',
+	'C':'-.-.',
+	'd':'-..',
+	'D':'-..',
+	'e':'.',
+	'E':'.',
+	'f':'..-.',
+	'F':'..-.',
+	'g':'--.',
+	'G':'--.',
+	'h':'....',
+	'H':'....',
+	'i':'..',
+	'I':'..',
+	'j':'.---',
+	'J':'.---',
+	'k':'-.-',
+	'K':'-.-',
+	'l':'.-..',
+	'L':'.-..',
+	'm':'--',
+	'M':'--',
+	'n':'-.',
+	'N':'-.',
+	'o':'---',
+	'O':'---',
+	'p':'.--.',
+	'P':'.--.',
+	'q':'--.-',
+	'Q':'--.-',
+	'r':'.-.',
+	'R':'.-.',
+	's':'...',
+	'S':'...',
+	't':'-',
+	'T':'-',
+	'u':'..-',
+	'U':'..-',
+	'v':'...-',
+	'V':'...-',
+	'w':'.--',
+	'W':'.--',
+	'x':'-..-',
+	'X':'-..-',
+	'y':'-.--',
+	'Y':'-.--',
+	'z':'--..',
+	'Z':'--..',
+	' ':' '
+}
+
+def error():
+	print("usage: " + sys.argv[0] + " <a-zA-Z string>")
+	exit(1)
+
+def is_morse(dictionary, strin):
+	for j in strin:
+		if not (j.isspace() or j.isalpha()):
+			error()
+		if not(dictionary[j]):
+			error()
+	return 1
```

Comment:
A more pythonic way of doing this would have been a oneliner:
```suggestion
def is_morse(strin):
    return strin.replace(" ").isalpha()
```

If you want to keep your function's form and make the dictionnary useful:
```suggestion
def is_morse(dictionary, strin):
    return "".join(c for c in strin if c in dictionary.keys()) != ""
```